### ISSUE TITLE
Palette selector does not fire change event on return to starting value

### ISSUE THREAD

**Issue by Netrilix** [bgrins/spectrum#326]
The palette selector fires change events every time a new color from the palette is clicked on, which is behavior I'm depending on to update a preview image below the selector (I know there's another issue open where people disagree that change should be fired on palette, but it works great this way for me).

However, if I start on a color (for example, black), any color I click on while the selector is still open will fire a change event to update my preview EXCEPT if I click on black again. If I click back to black (or whichever color was selected before the palette was opened), the change doesn't fire and my preview image stays at the last color chosen before black. The spectrum preview color changes properly, but the change event doesn't fire, so my own preview image below it doesn't.

To duplicate this behavior:

-Open the palette
-Choose any color (A) (change fires)
-Close the palette.
-Open the palette.
-Choose another color (B) (change fires)
-Optionally choose another color (C) (change fires)
-Choose the original color (A) again (change doesn't fire, but should)

I suspect there's code to prevent change from firing if the same color is chosen twice in a row, but instead of doing so, it's checking whether the color chosen matches the color active when the palette was opened rather than current active color.

**Comment by Netrilix** [comment]
For my own use case, I added a second parameter to `updateOriginalInput()` called `updateShownColor` (default `false`). If it's true, I update `colorOnShow` to `color` at the end of `updateOriginalInput()`. Then, when `paletteElementClick()` calls `updateOriginalInput()`, it passes this optional parameter as `true` to tell it to update.

Not knowing the code at all, and having only a cursory understanding of JavaScript, there's a good chance this change could have broken other things, but it seems to be working properly for me.

**Comment by Draeli** [comment]
I confirm the issue, I try in two cases :
1 : when I start with a default color in my input type color
2 : when I start without default color in my input type color but setting "color" value in configuration
In both case, when I return to the "default" color, 'change' event is not fire (change event in plugin or in original input DOM which is do usually)
I suspect the same problem describe above but if it was possible to have an "official" patch rather than override original lib ...

**Comment by bgrins** [comment]
> The palette selector fires change events every time a new color from the palette is clicked on, which is behavior I'm depending on to update a preview image below the selector (I know there's another issue open where people disagree that change should be fired on palette, but it works great this way for me).

The current plan is to remove the change event for palette selections as outlined in https://github.com/bgrins/spectrum/issues/250#issuecomment-66397975 and #269.  It's the most common source of confusion with this plugin.

For your use case, could you could wire up the same logic into both the move event and the hide event?  This would capture all changes (both palette and dragging color area), and should also handle reverted changes from the hide event.  Or is it that you are wanting to distinguish between palette clicks and dragging in the picker?

**Comment by bgrins** [comment]
After #405 (currently on master) I believe you can use a 'move' event if you want all changes happening while the picker is opened (including on the palette), or 'change' if you want a single event to fire once the picker has been closed.  Let me know if that doesn't cover your use case.
